pkg/event-manager/handlers_subscription.go

```diff
@@ -0,0 +1,173 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package eventmanager
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/entity-store"
+	"github.com/vmware/dispatch/pkg/event-manager/gen/models"
+	eventsapi "github.com/vmware/dispatch/pkg/event-manager/gen/restapi/operations/events"
+	subscriptionsapi "github.com/vmware/dispatch/pkg/event-manager/gen/restapi/operations/subscriptions"
+	"github.com/vmware/dispatch/pkg/trace"
+	"github.com/vmware/dispatch/pkg/utils"
+)
+
+func (h *Handlers) addSubscription(params subscriptionsapi.AddSubscriptionParams, principal interface{}) middleware.Responder {
+	defer trace.Trace("addSubscription")()
+
+	sp, _ := utils.AddHTTPTracing(params.HTTPRequest, "EventManager.addSubscription")
+	defer sp.Finish()
+
+	if err := params.Body.Validate(strfmt.Default); err != nil {
+		return eventsapi.NewEmitEventBadRequest().WithPayload(&models.Error{
+			Code:    http.StatusBadRequest,
+			Message: swag.String(fmt.Sprintf("error validating the payload: %s", err)),
+		})
+	}
+
+	e := subscriptionModelToEntity(params.Body)
+	e.Status = entitystore.StatusCREATING
```

Comment:
I think we should standardize on StatusINITIALIZED for initial state.  There will be a bigger discussion about this, as we are overloading the status field in general (trying to cram desired and actual state into a single field).

Reply:
I don't think each type can have exactly the same set of statuses (looking at kubernetes, each type has different status set, e.g. it makes sense for Pod to have `ImagePullBackOff` state, which makes no sense for deployment), but `StatusINITIALIZED` sounds good for the initial state.